predicators/approaches/pg3_approach.py

```diff
@@ -293,6 +293,84 @@ def _get_candidate_conditions(
         return conditions
 
 
+class _DeleteConditionPG3SearchOperator(_PG3SearchOperator):
+    """An operator that removes conditions from existing LDL rules."""
+
+    def get_successors(
+            self, ldl: LiftedDecisionList) -> Iterator[LiftedDecisionList]:
+        for rule_idx, rule in enumerate(ldl.rules):
+            for destination in ("pos", "neg", "goal"):
+                if destination == "pos":
+                    dest_set = set(rule.pos_state_preconditions)
+                elif destination == "neg":
+                    dest_set = set(rule.neg_state_preconditions)
+                else:
+                    assert destination == "goal"
+                    dest_set = set(rule.goal_preconditions)
+
+                for condition in dest_set:
```

Comment:
shouldn't this only be checked if the destination if "pos"?

Reply:
also, it looks like `dest_set` isn't getting used below here, maybe can remove it?